controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -246,17 +276,20 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 			// if we found more than one CephBlockPool, we can't determine which one to select, so error out
 			cbpItemsLen := len(cephBlockPoolList.Items)
 			if cbpItemsLen == 0 {
-				cbpNewName := fmt.Sprintf("cephblockpool-%s-%s", r.storageConsumer.Name, generateUUID())
+				cbpNewName := fmt.Sprintf("cephblockpool-profile-%s", profileName)
 				r.log.V(1).Info("no valid CephBlockPool found, creating new one", "CephBlockPool", cbpNewName)
 				r.cephBlockPool.Name = cbpNewName
+				r.cephBlockPool.Spec = r.getDefaultCephBlockPoolSpec()
+				addLabel(r.cephBlockPool, controllers.StorageProfileLabel, profileName)
```

Comment:
This is no according to the design. You are not supposed to create a new block pool, you should fail the reconcile if you cannot find an existing block pool to back up the claim.

Reply:
The more I worked on it, the more this design decision looks like a mistake. But sure, I'll remove it.

Reply:
Please see the last part of my comment here:
https://github.com/red-hat-storage/ocs-operator/pull/2478#discussion_r1530134814

Not agreeing with a design decision does not mean the design is a mistake. In this specific case, I can guess that you are missing the underlying point of moving away from managing storage concerns in provider mode.